### ISSUE TITLE
Clarify licensing of logo for unmodified builds from source for Linux distro

### ISSUE THREAD

**Issue by bgmilne** [MultiMC/MultiMC5#4102]
### Operating System

Linux

### Description of bug

I have packaged MultiMC (as `multimc`) for Mageia, currently only in the rolling-development-release referred to as Cauldron, primarily so my kids don't bother me to log into the Minecraft account using the official launcher (but also enjoying the features it has compared to the official launcher!) since it lost the ability to save the password.

You can find some details at the following links:
* https://madb.mageia.org/package/show/source/1/application/0/release/cauldron/name/multimc
* http://svnweb.mageia.org/packages/cauldron/multimc/current/SPECS/multimc.spec?revision=1745372&view=markup
* https://repology.org/project/multimc/versions

However, I haven't added the icon, so the menu entry currently is a bit dull, since https://multimc.org/#Download says:
> MultiMC is released under the Apache 2 license, which gives you the full access to the source code. What it doesn't give you is the right to use the name (MultiMC) or logo (infinity symbol on the side view of a block of grass) in any custom builds or forks of the project.

> However, as long as you only redistribute the **unmodified binary builds** from multimc.org, you may use the following files for the purpose of linux packaging and/or giving the user(s) a shortcut for launching the application.

According to my interpretation of the current text, we may not use the logo.

However, other Linux distributions (any based on Arch) seem to use the logo, see e.g. 
* https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=multimc-git#n66
* https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=multimc5#n76
* https://gitweb.gentoo.org/repo/gentoo.git/tree/games-action/multimc/multimc-0.6.13.ebuild#n71

I would like to make multimc available for the current stable release (Mageia 8), but I would prefer to do so with in icon in the menu, but without violating the terms of use for the logo or other marks.

I see some discussion of the use of the name MultiMC also not being permitted (e.g. in https://github.com/MultiMC/MultiMC5/issues/3183). If this is required too, I will rename it (and replace all references in the spec file that end up being user-visible), but it could also preclude us from having the [URL in the package](http://svnweb.mageia.org/packages/cauldron/multimc/current/SPECS/multimc.spec?revision=1745372&view=markup#l7).

I personally avoid installing any 3rd-party software on my Linux machines (if I'm going to compile from source, I may as well package it), and I think it may be beneficial to your project to reach more users to allow Linux distros to ship binaries built from unmodified source, if a compromise can be reached (besides renaming the package arbitrarily with different distros using different names).

### Steps to reproduce

Attempt to package a fully working package for a Linux distro with an attractive menu entry.

### Suspected cause

License terms for logo (and name?) insufficiently clear compared to usage in other distributions.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by kb-1000** [comment]
As far as I'm aware, you'd have to use the binaries from https://multimc.org (using something similar to https://github.com/MultiMC/MultiMC5/blob/develop/launcher/package/ubuntu/multimc/opt/multimc/run.sh)
The two AUR packages you mentioned are unofficial, and absolutely not recommended to use.
I don't have any information about that Gentoo package, but it's outdated and thus probably not recommended either.

**Comment by peterix** [comment]
You do not have permission to build and redistribute the MultiMC codebase without changing the name, logo, and all API keys and internal service endpoints to your own.

**Comment by peterix** [comment]
In other words: Please use the binary builds as available on multimc.org

**Comment by peterix** [comment]
Or maintain a fully separate fork and assume all required responsibilities

**Comment by Schmensch** [comment]
Well but if he would just package a script that then downloads the newest MultiMC-binary from https://multimc.org, that would be ok? Because then he's just using the unmodified binaries from the website.

**Comment by peterix** [comment]
That's what the deb package does, and what multimc-bin in the AUR does. It's definitely a good way to do it.

**Comment by the-maldridge** [comment]
@peterix I want to double check that the intention here is to drop MultiMC from all distribution by Linux distros.  I'm over with Void Linux and we're in the process of expunging MultiMC from our repositories as this seems to be in line with your wishes, but I do want to double check that your/the project's intent is to discontinue broad Linux distribution.

**Comment by anoraktrend** [comment]
Just dropping a related [link](https://en.m.wikipedia.org/wiki/Mozilla_software_rebranded_by_Debian) to historical information about a similar thing occuring in 2006 that resulted in the creation of "IceWeasel", which is still around today. No real point, just noticed a similarity to that and this.

**Comment by Schmensch** [comment]
@the-maldridge it seems like MultiMC is shifting to be more like Chrome and VSCode:

The project is open-source, then it gets forked internally and the proprietary branding and API keys are added. If you build it from source, you get an unbranded version without some features.  (e. g. in case of chrome / chromium you lose syncing capabilities)

**Comment by peterix** [comment]
Yep. Pretty much. Sadly, the pirate forks and the terms and conditions of the ms identity platform don't really give me other options.

On top of that, it's kinda hard to support weird builds and packaging.

**Comment by the-maldridge** [comment]
Hmm, that is unfortunate as I suspect it will lead to the loss of most Linux users.  @andersmmg I'm well aware of the IceWeasel history, which is why I wanted to learn more about the reasoning here.  Given that Mozilla has largely gone back on the reasoning that led to IceWeasel in the first place due to the problems it caused I really wanted to know if the same reasoning was being used here.  It sounds like it is.

@peterix thanks for indulging my curiosity.  I have to admit I neither agree with nor understand your viewpoint on this matter, but I really do appreciate getting to hear the reasoning directly.

**Comment by peterix** [comment]
In any case, unmodified builds will not get you 'MultiMC' in the future. More like 'Launcher', and you will have to register your own application with Microsoft and agree to their terms and conditions:
https://docs.microsoft.com/en-us/legal/microsoft-identity-platform/terms-of-use

Wording like `All activities that occur using your Access Credentials are your responsibility` is ... a bit too much.

I basically have several options:
- Agree with MS terms and do what I'm doing, even though I think the terms are insane and detached from reality.
- Use the MS identity stuff while ignoring its terms and possibly getting in conflict with MS, while getting the users in the crossfire.
- Kill the project and go watch the grass grow.
- Go full pirate.

There might be some others, but what I'm doing is the only way forward that, from my POV, doesn't involve legal and existential risks.

**Comment by peterix** [comment]
As for void linux...

Looking at the analytics, there are no void linux users.

**Comment by peterix** [comment]
It's possible they did not opt in to be counted... But still, in general, about 20% of users do (compared to traffic to servers MultiMC depends on).

**Comment by the-maldridge** [comment]
You're certainly between a rock and a hard place.  I can respect that and I encourage you to lodge your frustration with Microsoft.  They won't do anything, but its still important to have the data point that what they're doing is disruptive and indefensible.  

As for your metrics, I would not be surprised at all if Void users had a 100% opt out rate/blocked tracking servers based on the profile of users who select Void as their preferred distro.

**Comment by peterix** [comment]
Yeah. I wouldn't be surprised about that. It's opt in and requires user decision to activate for a reason.

In any case, the binary builds should work just fine as long as the necessary ABIs exist on the system. Which is mostly a very old version of glibc (think 2014) and Qt 5.4 or newer.

**Comment by Xiretza** [comment]
Hi, maintainer of [`multimc5`](https://aur.archlinux.org/packages/multimc5) chiming in.

Just to make sure I understand the situation correctly: Minecraft is deprecating Mojang accounts in favor of Microsoft accounts, so MultiMC has to support Microsoft authentication to stay functional. Due to the horrible (or at least horribly unclear) terms of use for Microsoft's authentication API tokens, you don't want anyone using your tokens for anything but the prebuilt MultiMC binaries.

That much makes sense to me. My personal solution here would be to add an easy method for defining an API token at compile time (such as a cmake variable), so users that want to build from source can just use their own tokens.

Instead, your demand seems to be that anyone building the software from source (and thus with a different API token) must completely rebrand it. While I don't understand this choice, I also don't really care if the package I maintain is called `multimc5` or `generic-minecraft-launcher-that-definitely-isnt-multimc`. Just make that the default and I'll rename the package accordingly.

**Comment by kb-1000** [comment]
> Part of the reason for using the Apache license is that we don't want people using the "MultiMC" name when redistributing the project. **This means people must take the time to go through the source code and remove all references to "MultiMC", including but not limited to the project icon and the title of windows, (no MultiMC-fork in the title).**
> 
> Apache covers reasonable use for the name - a mention of the project's origins in the About dialog and the license is acceptable. However, it should be abundantly clear that the project is a fork without implying that you have our blessing.

If you want to continue maintaining that package, you'll have to rename it. And `generic-minecraft-launcher-that-definitely-isnt-multimc` will not do. In addition to that, you'll have to have your own place for support (or none at all), and host your own meta server (which includes hosting a few files referenced by the meta repository and the launcher code that are hosted on multimc.org). If you want Microsoft authentication to work, you must also have your own client id.

You also have to comply with the other parts of the license, of course.

Something like `multimc-bin` or the .deb/.rpm packages we have are not redistributing the project, but rather using official binaries using download scripts and thus are fine.

**Comment by Schmensch** [comment]
I don't know if I'm suggesting too much, but if I would maintain MultiMC, I would

- change the branding inside the launcher to something generic 
- change the name of this repo to something generic.
- change the license to MIT, to allow people to redistribute the generic, unbranded binaries
- make a internal branch with a different / proprietary license, and add branding, api keys, ... back in and then redistribute that binary.

basically doing the exact same thing VSCode did:

- The binaries on https://code.visualstudio.com have a proprietary license, proprietary branding, telemetry and api keys. 
- If you build it from source, you get an generic version named `code-oss` instead, that has no branding, no syncing capabilities, because no API keys and no telemetry.
- `code-oss` is licensed under MIT, while VSCode is licensed under a very proprietary Microsoft License.
- Because `code-oss` is licensed under MIT, you can repackage and redistribute as much as you like.

TLDR:

Make branding of MultiMC proprietary and release the unbranded version under MIT so that there aren't as much licensing violations.

**Comment by Schmensch** [comment]
Also, but this is optional, easy options to add your own API keys / branding during compile time would be nice.

**Comment by jamierocks** [comment]
I don't see any need to relicense the project?, and doing so would be encumbered with the task of getting permission from contributors.

**Comment by Zetabite** [comment]
Plus half of the things mentioned are already on the agenda of MMC, and really arent dependent on the MIT lisensce.

**Comment by Schmensch** [comment]
> I don't see any need to relicense the project?, and doing so would be encumbered with the task of getting permission from contributors.

Actually I think you only have to do this if you're using copyleft licenses, like GPLv3. Copyright licenses like MIT, Apache or BSD can be changed if you desire to do so.

**Comment by theofficialgman** [comment]
> In any case, unmodified builds will not get you 'MultiMC' in the future. More like 'Launcher', and you will have to register your own application with Microsoft and agree to their terms and conditions:
> https://docs.microsoft.com/en-us/legal/microsoft-identity-platform/terms-of-use
> 
> Wording like `All activities that occur using your Access Credentials are your responsibility` is ... a bit too much.
> 
> I basically have several options:
> 
> * Agree with MS terms and do what I'm doing, even though I think the terms are insane and detached from reality.
> * Use the MS identity stuff while ignoring its terms and possibly getting in conflict with MS, while getting the users in the crossfire.
> * Kill the project and go watch the grass grow.
> * Go full pirate.
> 
> There might be some others, but what I'm doing is the only way forward that, from my POV, doesn't involve legal and existential risks. Or significantly changing the nature of the project that would affect all users.

there is one option you have not considered thus far which could potentially solve all problems

you could include a compiled library for the secrets (a shared library with the client id contained for every desired system/architecture) in the multimc5 repo. Building MultIMC5 could still be by the user if they so desire, it would just involve linking to a pre-compiled and included shared library. This way the client id somewhat obfuscated  (depending on how you compile the library)

**Comment by kb-1000** [comment]
Answer has been stated above and elsewhere publicly, so I see this as resolved.
closing.

**Comment by peterix** [comment]
I don't think that would do.

If you actually read the terms of the identity platform, it imposes all kinds of responsibilities on me that are extremely unusual in any sort of open source project. Pretty much any open source / free software license says that the code comes with no guarantees.

It's fundamentally incompatible.

If I just give you a secrets library to link with, you can take that and link with it. Let's say you then also change the launcher code to let users play without owning the game.

The effect of that is:
- You have made something that breaks multiple clauses in that agreement, in my name.
- It looks like MultiMC, which is misleading.
- I helped you doing it by giving you all the 'secrets' and branding on a silver platter.

So no, this is not going to happen.

**Comment by peterix** [comment]
It's not that I couldn't trust you or the package maintainers of any linux distro out there. Millions of users trust distro package maintainers.

But if I do so publicly and without any sort of strings attached, I am letting anyone to use the secrets and I am trusting every single one of them to not do something stupid.

**Comment by peterix** [comment]
Theoretically, I could set up a company and give out the 'secrets' under NDA... But that's an immense amount of overhead.

**Comment by anoraktrend** [comment]
Alternatively, i've just started MonoMC as a project to get the Iglunix launcher its own branding. Ella, (the head dev of Iglunix) has their own ideas as to how to work around this issue, but I both understand and sympathize with your own issues. MonoMC is just branding right now, but the branding itself is under a different license, particularly CC0 for the existing art.

**Comment by C0rn3j** [comment]
@peterix Until there is a better solution implemented, would you consider adding a makefile option to add your own Microsoft API key for now?

**Comment by phit** [comment]
@C0rn3j https://github.com/MultiMC/MultiMC5/tree/develop/notsecrets

**Comment by lilydjwg** [comment]
Hi, if only the secrets are what prevent downstream packaging, could you make the secrets a required configuration so that if not compiled in, users can be hinted to provide themselves, and have the project name unchanged?
